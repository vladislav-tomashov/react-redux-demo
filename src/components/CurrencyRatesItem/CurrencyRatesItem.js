import React from "react";
import numeral from "numeral";

import "./CurrencyRatesItem.scss";

const getFormattedRate = rate => {
  if (isNaN(parseFloat(rate)) || !isFinite(rate)) {
    return String(rate);
  }
  return numeral(rate).format("0.0000");
};

const CurrencyRatesItem = ({ currency, rate }) => (
  <div className="CurrencyRatesItem">
    <span>{currency}</span>
    <span>{getFormattedRate(rate)}</span>
  </div>
);

export default CurrencyRatesItem;
