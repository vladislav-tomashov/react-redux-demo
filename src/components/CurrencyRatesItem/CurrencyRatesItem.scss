@import "../../styles/settings";

.CurrencyRatesItem {
  display: flex;
  justify-content: space-between;
  text-align: left;
  line-height: 1.3;
  border-bottom-style: dotted;
  border-bottom-width: 1px;
  &:hover {
    background-color: lighten($light-blue, 10%);
    color: $black;
    font-weight: 600;
  }
}

.CurrencyRatesItem > * {
  padding: 0 $s-size;
}
