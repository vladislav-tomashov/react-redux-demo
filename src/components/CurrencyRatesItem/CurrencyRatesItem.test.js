import React from "react";
import { shallow } from "enzyme";
import CurrencyRatesItem from "./CurrencyRatesItem";

let wrapper;

describe("CurrencyRatesItem component", () => {
  beforeEach(() => {
    const props = {
      currency: "ABC",
      rate: 1
    };
    wrapper = shallow(<CurrencyRatesItem {...props} />);
  });

  test("should render CurrencyRatesItem correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
