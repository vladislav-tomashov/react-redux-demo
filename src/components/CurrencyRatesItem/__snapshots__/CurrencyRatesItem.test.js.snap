// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyRatesItem component should render CurrencyRatesItem correctly 1`] = `
<div
  className="CurrencyRatesItem"
>
  <span>
    ABC
  </span>
  <span>
    1.0000
  </span>
</div>
`;
